src/platform/forms-system/src/js/components/FormNav.jsx

```diff
@@ -98,6 +100,43 @@ export default function FormNav(props) {
     : `Step ${currentChapterDisplay} of ${chaptersLengthDisplay}: ${chapterName ||
         ''}`;
 
+  const handleFocus = async () => {
+    let root = document.querySelector('#react-root');
+    if (formConfig.v3SegmentedProgressBar) {
+      // Need to provide shadowRoot for focusing on shadow-DOM elements
+      const shadowHost = await querySelectorWithShadowRoot(
+        'va-segmented-progress-bar',
+      );
+      root = shadowHost.shadowRoot;
+    }
+
+    return () => {
+      // Check main toggle to enable custom focus; the unmounting of the page
+      // before the review & submit page may cause the customScrollAndFocus
+      // function to be called inadvertently
+      if (
+        !(
+          page.chapterKey === 'review' ||
+          window.location.pathname.endsWith('review-and-submit')
+        )
+      ) {
+        if (formConfig.useCustomScrollAndFocus) {
+          customScrollAndFocus(page.scrollAndFocusTarget, index);
+        } else {
+          waitForRenderThenFocus(defaultFocusSelector, root, 400);
+        }
+      } else {
+        // h2 fallback for review page
+        focusByOrder([defaultFocusSelector, 'h2'], root);
+      }
+    };
+  };
+
+  // Handle focus on mount
+  useEffect(() => {
+    handleFocus();
+  }, []);
```

Comment:
Is this necessary? since you already have a separate use effect for this, which will also trigger on mount + when its state changes